recipes/historyserver.rb

```diff
@@ -0,0 +1,87 @@
+my_ip = my_private_ip()
+my_public_ip = my_public_ip()
+
+
+completed_jobs_dir = "#{node['flink']['historyserver']['remote_dir']}"
+
+hops_hdfs_directory completed_jobs_dir do
+    action :create_as_superuser
+    owner node['flink']['user']
+    group node['flink']['group']
+    mode "1733"
+end
+
+
+deps = ""
+if exists_local("hops", "nn")
+  deps = "namenode.service"
+end
+service_name="flinkhistoryserver"
+
+service service_name do
+    provider Chef::Provider::Service::Systemd
+    supports :restart => true, :stop => true, :start => true, :status => true
+    action :nothing
+  end
+
+case node['platform_family']
+when "rhel"
+  systemd_script = "/usr/lib/systemd/system/#{service_name}.service"
+else
+  systemd_script = "/lib/systemd/system/#{service_name}.service"
+end
+
+ruby_block "hadoop_glob" do
+  block do
+    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
+    node.default['hadoop_glob'] = shell_out("#{node['hops']['bin_dir']}/hadoop classpath --glob  | tr -d '\n'").stdout
+  end
+  action :create
+end
+
+template node['flink']['historyserver']['environment'] do
+    source "historyserver.env.erb"
+    owner node['flink']['user']
+    group node['flink']['group']
+    mode 0750
+    variables({
+      :hadoop_glob=> lazy { node['hadoop_glob'] },
+      :flink_log_dir=> node['flink']['historyserver']['logs']
```

Comment:
You can avoid passing the the `flink_log_dir` as variable, just use `<%= node['flink']['historyserver']['logs'] %>`. Keeps the recipes more concise and readable